docs/source/task_guides/semantic_segmentation_lora.mdx

```diff
@@ -0,0 +1,445 @@
+```
+
+At this point you can check with the `print_trainable_parameters` helper function that all 100% parameters in the base 
+model (aka `model`) are trainable.
+
+## Wrap the base model as a PeftModel for LoRA training
+
+To leverage the LoRa method, you need to wrap the base model as a `PeftModel`.  This involves two steps:
+
+1. Defining LoRa configuration with `LoraConfig`
+2. Wrapping the original `model` with `get_peft_model()` using the config defined in the step above.
+
+```python
+from peft import LoraConfig, get_peft_model
+
+config = LoraConfig(
+    r=32,
+    lora_alpha=32,
+    target_modules=["query", "value"],
+    lora_dropout=0.1,
+    bias="lora_only",
+    modules_to_save=["decode_head"],
+)
+lora_model = get_peft_model(model, config)
+print_trainable_parameters(lora_model)
+```
+
+Let's review the `LoraConfig`. To enable LoRA technique, we must define the target modules within `LoraConfig` so that 
+`PeftModel` can update the necessary matrices. Specifically, we want to target the `query` and `value` matrices in the 
+attention blocks of the base model. These matrices are identified by their respective names, "query" and "value". 
+Therefore, we should specify these names in the `target_modules` argument of `LoraConfig`.
+
+After we wrap our base model `model` with `PeftModel` along with the config, we get 
+a new model where only the LoRA parameters are trainable (so-called "update matrices") while the pre-trained parameters 
+are kept frozen. These include the parameters of the randomly initialized classifier parameters too. This is NOT we want 
+when fine-tuning the base model on our custom dataset. To ensure that the classifier parameters are also trained, we 
+specify `modules_to_save`. This also ensures that these modules are serialized alongside the LoRA trainable parameters 
+when using utilities like `save_pretrained()` and `push_to_hub()`.
+
+In addition to specifying the `target_modules` within `LoraConfig`, we also need to specify the `modules_to_save`. When 
+we wrap our base model with `PeftModel` and pass the configuration, we obtain a new model in which only the LoRA parameters 
+are trainable, while the pre-trained parameters and the randomly initialized classifier parameters are kept frozen. 
+However, we do want to train the classifier parameters. By specifying the `modules_to_save` argument, we ensure that the 
+classifier parameters are also trainable, and they will be serialized alongside the LoRA trainable parameters when we 
+use utility functions like `save_pretrained()` and `push_to_hub()`.
+
+Let's review the rest of the parameters:
+
+- `r`: The dimension used by the LoRA update matrices.
+- `alpha`: Scaling factor.
+- `bias`: Specifies if the `bias` parameters should be trained. `None` denotes none of the `bias` parameters will be trained.
+
+`r` and `alpha` together control the total number of final trainable parameters when using LoRA, giving you the flexibility
```

Comment:
```suggestion
```

Reply:
removing this line as it isn't correct. 
Actually, the delta activations from LoRA are scaled as per alpha/r. So, higher alpha means that higher weightage is given to the LoRA activations and as such the gradients are also scaled accordingly. MAM paper uses LoRA and Scaled Parallel Adapter with alpha being 4X compared to r. The LoRA paper itself uses alpha ranging from 1X-8X that of r.

Alpha doesn't change the number of trainable parameters, rather functions as a scaling factor.

Reply:
Thank you for the explanation!